src/hooks/useSession.tsx

```diff
@@ -0,0 +1,211 @@
+import React, {
+  createContext,
+  useEffect,
+  useContext,
+  useState,
+  useMemo,
+  useCallback,
+} from 'react';
+import { Account, Subject, Entry, User, SubjectWithProject } from '../types';
+import { useAsyncStorage } from './useAsyncStorage';
+import { useHttpClient } from './useHttpClient';
+import { APIClient } from '@lifeomic/one-query';
+import { RestAPIEndpoints } from '../types/rest-types';
+import { useAuth } from './useAuth';
+
+type Configuration = {
+  account: string;
+  subjects: SubjectWithProject[];
+};
+
+type UserConfiguration = {
+  sessionTime: string;
+  user: User;
+  accounts: Account[];
+  configurations: Configuration[];
+};
+
+type UserConfigurationsContextProps = {
+  userConfiguration: UserConfiguration;
+  isLoaded: boolean;
+  clearSession: () => void;
+};
+
+enum Session {
+  placeholder = 'placeholder',
+}
+
+const configurationPlaceholder: UserConfiguration = {
+  user: {
+    id: Session.placeholder,
+    profile: {
+      email: Session.placeholder,
+    },
+  },
+  configurations: [],
+  accounts: [],
+  sessionTime: new Date().toISOString(),
+};
+
+export const SessionContext = createContext({
+  userConfiguration: configurationPlaceholder,
```

Comment:
Is the thought that we might add more of these types of objects to the context over time?

Reply:
I've at least envisioned using the result of appConfig to determine if trackers should be fetched and stored in session for the sake of reducing/removing the loading time for the track tile on the Home Screen.